charm-slurmd/src/interface_slurmd.py

```diff
@@ -132,6 +137,26 @@ def slurmctld_port(self):
         """Get slurmctld port."""
         return self._stored.slurmctld_port
 
+    @property
+    def node_inventory(self):
```

Comment:
```suggestion
    def node_inventory(self) -> dict:
```

I think it makes development easier if we start adding type hints. What do you think about it?